### PR TITLE
refactor: Migrate MultipartEncoder to RequestsBackend

### PR DESCRIPTION
**Pull request by lmilbaum**

### CODE REVIEW

**Comment by lmilbaum** [comment]
@nejch Could you please place a lated `ok-to-test` on this PR? I want to verify the error is not caused by a flaky test.

**Comment by nejch** [comment]
> @nejch Could you please place a lated `ok-to-test` on this PR? I want to verify the error is not caused by a flaky test.

@lmilbaum you shoul have Triage rights now (check your email if not), so you should be able to do so yourself in the future.

**Comment by lmilbaum** [comment]
> > @nejch Could you please place a lated `ok-to-test` on this PR? I want to verify the error is not caused by a flaky test.
> 
> @lmilbaum you shoul have Triage rights now (check your email if not), so you should be able to do so yourself in the future.

yay. Thank you for trusting me.

**Comment by lmilbaum** [comment]
> @nejch Could you please place a lated `ok-to-test` on this PR? I want to verify the error is not caused by a flaky test.

yup. It was a flaky test.

File: gitlab/http_backends/requests_backend.py [hunk 1]
```diff
@@ -39,6 +39,38 @@ def __init__(self, session: Optional[requests.Session] = None) -> None:
     def client(self) -> requests.Session:
         return self._client
 
+    @staticmethod
+    def prepare_send_data(
```
Comment (nejch) [inline gitlab/http_backends/requests_backend.py:43]: I think we should make these things private somehow before we release this, maybe not here but by making the client backends private or so. Not urgent for now.

**Pull request merged**